### PR TITLE
Fixes NullPointerExceptions due to callbacks in MainActivity

### LINKED ISSUE #891: Upon closing all the tabs and pressing back button , app crashes.
**Describe the bug**

App crashes when all tabs are closed and back button is pressed

**Expected behavior**

The app should just exit without crashing.

**Steps to reproduce the behavior:**

1. Open two or more tabs are open.
2. Close all tabs using CloseAllTabs or swiping up for all tabs.
3. Press back button.

**Screenshots**

kiwix crash 3 gif

**Environment**
- Device : Redmi 4
- OS version : Android 7.1.2

**Logs**

2018-11-12 12:22:03.996 24644-24644/org.kiwix.kiwixmobile E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.kiwix.kiwixmobile, PID: 24644
    java.lang.NullPointerException: Attempt to invoke virtual method 'void org.kiwix.kiwixmobile.main.AnimatedProgressBar.setProgress(int)' on a null object reference
        at org.kiwix.kiwixmobile.main.MainActivity.webViewProgressChanged(MainActivity.java:1775)
        at org.kiwix.kiwixmobile.main.KiwixWebChromeClient.onProgressChanged(KiwixWebChromeClient.java:33)
        at xq.a(SourceFile:47)
        at alW.handleMessage(SourceFile:81)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:163)
        at android.app.ActivityThread.main(ActivityThread.java:6228)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:904)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:794)

**Request**
Please assign me for this issue.

### PR DESCRIPTION
**Pull request by sonusourav**
Fixes #891 #895 

Issues #891 and #895 have a common root to callbacks which gives NullPointerExceptions.
Changes:
1. Added a function named CheckNull. 
2. CheckNull checks for NullPointerException for views in MainActivity.

**Screenshots**
checknull

### CODE REVIEW

**Comment by soloturn** [comment]
is this overlapping with #1038 ?

**Comment by soloturn** [comment]
@sonusourav @abdulwd @prajurock is there a design problem as we need such null pointer checks all over the place? i fell over a comment saying:
"Once setContentView() has been called, you will never get a null View provided you are looking in the correct layout and the View exists in that layout."

https://stackoverflow.com/questions/16437050/checking-if-a-view-null-before-setting-onclicklistener

File: app/src/main/java/org/kiwix/kiwixmobile/main/MainActivity.java [hunk 1]
```diff
@@ -1950,4 +1961,8 @@ private void searchFiles() {
       fileSearch.scan(sharedPreferenceUtil.getPrefStorage());
     }
   }
+
+  public boolean checkNull(View view) {
+    return view != null;
+  }
```
Comment (harshika-arya) [inline app/src/main/java/org/kiwix/kiwixmobile/main/MainActivity.java]: In my opinion, encapsulating such small functionality in a separate function is overkill. You are anyways going to write `if(checkNull(view_name)){}`. Why not just `if(view_name != null)`. Every function call has an overhead and hence I am not in favor of it. If readability is your main intention, then we might leave it like you have done but I find explicit check equally readable.

**Comment by harshika-arya** [comment]
@soloturn Hi, So I went through the codebase a bit more after going through the Stackoverflow link. I must say that I might be wrong since I started yesterday only and hence am relatively new to this project. 

I think the problem is indeed a design issue.

1. We open the SettingsActivity to change settings.
2. We change a few things which requires us to restart the MainActivity. The `onActivityResult`(Line 1417 - MainActivity.java) method handles the callback using this snippet

```Java
case REQUEST_PREFERENCES:
        if (resultCode == RESULT_RESTART) {
          startActivity(new Intent(MainActivity.this, MainActivity.class));
          finish();
        }
        ... more code.
```

3. The Activity is recreated. While the Activity is getting recreated(not yet finished), the `KiwixWebChromeClient` uses the callback to update the view using methods such as `webViewProgressChanged`. Now because the views are still being recreated, we get a NPE and hence crash.

**Comment by soloturn** [comment]
@harshika-arya thanks for having a look! how would you recommend to implement this?

**Pull request merged**